src/shared/components/LomFieldsInput/LomFieldsInput.tsx

```diff
@@ -0,0 +1,170 @@
+import { Form, FormGroup, Spacer, TagsInput } from '@viaa/avo2-components';
+import { Avo } from '@viaa/avo2-types';
+import { differenceWith, isEmpty, isEqual, isNil, map, without } from 'lodash-es';
+import React, { FC, useEffect, useState } from 'react';
+
+import { CustomError } from '../../helpers';
+import useTranslation from '../../hooks/useTranslation';
+import { LomService } from '../../services/lom.service';
+import { LomFieldsByScheme } from '../../types/lom';
+
+import { mapLomFieldsToOptions, mapOptionsToLomFields } from './LomFieldsInput.helpers';
+
+type LomFieldsInputProps = {
+	contexts: Avo.Lom.LomField[];
+	educationLevels: Avo.Lom.LomField[];
+	subjects: Avo.Lom.LomField[];
+	themes: Avo.Lom.LomField[];
+	onChange: (lomFields: Partial<LomFieldsByScheme>) => void;
+};
+
+const LomFieldsInput: FC<LomFieldsInputProps> = ({
+	contexts,
+	educationLevels,
+	subjects,
+	themes,
+	onChange,
+}) => {
+	const { tText } = useTranslation();
+	const [educationLevelsOptions, setEducationLevelsOptions] = useState<Avo.Lom.LomField[]>([]);
+	const [subjectsOptions, setSubjectsOptions] = useState<Avo.Lom.LomField[]>([]);
+	const [themesOptions, setThemesOptions] = useState<Avo.Lom.LomField[]>([]);
```

Comment:
don't use state to store stuff that needs to react to changes in parent state
just use a getEducationLevelOptions() function that does this dyanmically on each render
if it is too slow we can wrap it in a useMemo hook